railgun/apis/configuration/v1alpha1/ingress_class_params_types.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2021 Kong, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&IngressClassParams{}, &IngressClassParamsList{})
+}
+
+//+kubebuilder:object:root=true
+
+// IngressClassParamsList contains a list of IngressClassParams
+type IngressClassParamsList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []IngressClassParams `json:"items"`
+}
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//+kubebuilder:object:root=true
+//+kubebuilder:storageversion
+
+// IngressClassParams is the Schema for the ingressclassparams API
+type IngressClassParams struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec IngressClassParamsSpec `json:"spec,omitempty"`
+}
+
+// IngressClassParamsSpec defines the desired state of IngressClassParams
+type IngressClassParamsSpec struct {
+	// Offload load-balancing to kube-proxy or sidecar
+	ServiceUpstream bool `json:"serviceUpstream,omitempty"`
```

Comment:
Should we make this optional? Should we set a default value here? There are 2 ways of doing that that I can think of:

- making this field `*bool`
- specifying a default value in the CRD

Which one is better in your opinion?